source/game/render/AnmMgr.hh

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "game/system/ResourceManager.hh"
+
+#include <abstract/g3d/ResFile.hh>
+
+#include <list>
+
+namespace Render {
+
+class DrawMdl;
+
+enum class AnmType : s32 {
+    Empty = -1,
+    Chr = 0,
+    Clr = 1,
+    Srt = 2,
+    Pat = 3,
+    Shp = 4,
+    Max = 5,
+};
+
+class AnmNode {
+public:
+    AnmNode(Abstract::g3d::AnmObjChrRes anmObjChrRes, AnmType anmType, size_t idx)
+        : m_anmObjChrRes(anmObjChrRes), m_anmType(anmType), m_idx(idx) {}
+
+    [[nodiscard]] f32 frameCount() const {
+        return static_cast<f32>(m_anmObjChrRes.frameCount());
+    }
+
+private:
+    Abstract::g3d::AnmObjChrRes m_anmObjChrRes;
+    AnmType m_anmType;
+    size_t m_idx;
+};
```

Comment:
Fine if we don't want to do inheritance for now, but referencing AnmObjChrRes directly should imply this is an AnmNodeChr.

Reply:
So you're suggesting `class AnmNode : public Abstract::g3d::AnmObjChrRes`?